lib/kitchen/directions/bake_references/v3.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module Kitchen::Directions::BakeReferences
+  class V3
+    renderable
+
+    def bake(book:, metadata_source:)
+      @metadata = metadata_source.children_to_keep.copy
+      @klass = 'references'
+      @uuid_prefix = '.'
+      @title = I18n.t(:references)
+
+      book.chapters.each do |chapter|
+        chapter.pages.each do |page|
+          page.references.each do |reference|
+            title = reference.titles.first
+            title.name = 'h2'
+            title['data-type'] = 'document-title'
+            title['id'] = page.title.copied_id
+            title.replace_children(with:
+              <<~HTML.chomp
+                <span class="os-number">#{page.title.children[0].text}</span>
+                <span class="os-divider"> </span>
+                <span class="os-text" data-type="" itemprop="">#{page.title_text} </span>
+              HTML
+            )
```

Comment:
Did you see that there's a [v2](https://github.com/openstax/kitchen/blob/main/lib/kitchen/directions/eoc_section_title_link_snippet.rb#L19) that might help here? Looks like the only difference is `h2` or `h3` in the title element, but that could be modified after the fact.